youtube_dl/extractor/deezer.py

```diff
@@ -8,84 +8,192 @@
     ExtractorError,
     int_or_none,
     orderedSet,
+    bytes_to_intlist,
+    intlist_to_bytes,
 )
 
+from hashlib import md5
+from ..aes import aes_ecb_encrypt
+
+from binascii import b2a_hex
+from ..compat import compat_ord, compat_chr
+
+
+def md5hex(data):
+    """ return hex string of md5 of the given string """
+    return md5(data).hexdigest().encode('utf-8')
+
+
+def calcurlkey(songid, md5origin, mediaver=4, fmt=1):
+    """ Calculate the deezer download url given the songid, origin and media+format """
+    data = b'\xa4'.join(_.encode("utf-8") for _ in [md5origin, str(fmt), str(songid), str(mediaver)])
+    data = b'\xa4'.join([md5hex(data), data]) + b'\xa4'
+    if len(data) % 16:
+        data += b'\x00' * (16 - len(data) % 16)
+    enc = aes_ecb_encrypt(bytes_to_intlist(data), bytes_to_intlist(b"jo6aey6haid2Teih"))
+    return b2a_hex(intlist_to_bytes(enc)).decode('utf-8')
+
+
+def calcblowfishkey(songid):
+    """ Calculate the Blowfish decrypt key for a given songid """
+    h = md5hex(str(songid).encode('ascii'))
+    key = b"g4el58wc0zvf9na1"
```

Comment:
How did you obtain this key?

Reply:
It must be available somewhere in the browser but I could not find it. Installing a tool like [Firebug ](https://getfirebug.com/)is a good start to find it.

The @deezerdl user [has been created for this pull request](https://github.com/deezerdl?tab=overview&from=2017-12-01&to=2017-12-31), he will probably never answer...